### ISSUE TITLE
Uncaught TypeError: join(): Argument #2 ($array) must be of type ?array, string given

### ISSUE THREAD

**Issue by mikeg-de** [ianhampton/Tealium-Wordpress-Plugin#7]
WordPress version 6.1.1
Active theme: Bright (version 1.0.1)
Current plugin: Tealium (version 2.1.15) PHP version 8.0.25
 
```
Error Details
=============
An error of type E_ERROR was caused in line 24 of the file /data/www/bright-carrosserie.be/www/wp-content/plugins/tealium/tealium.options.php. Error message: Uncaught TypeError: join(): Argument #2 ($array) must be of type ?array, string given in /data/www/bright-carrosserie.be/www/wp-content/plugins/tealium/tealium.optio
ns.php:24

Stack trace:
#0 /data/www/bright-carrosserie.be/www/wp-content/plugins/tealium/tealium.options.php(24): join()
#1 /data/www/bright-carrosserie.be/www/wp-content/plugins/tealium/tealium.options.php(85): tealiumFormatAsName()
#2 /data/www/bright-carrosserie.be/www/wp-content/plugins/tealium/tealium.options.php(261): tealiumGenerateBulkDataSourceList()
#3 /data/www/bright-carrosserie.be/www/wp-content/plugins/tealium/tealium.php(82): include('...')
#4 /data/www/bright-carrosserie.be/www/wp-includes/class-wp-hook.php(308): options_page_tealium()
#5 /data/www/bright-carrosserie.be/www/wp-includes/class-wp-hook.php(332): WP_Hook->apply_filters()
#6 /data/www/bright-carrosserie.be/www/wp-includes/plugin.php(517): WP_Hook->do_action()
#7 /data/www/bright-carrosserie.be/www/wp-admin/admin.php(259): do_action()
#8 /data/www/bright-carrosserie.be/www/wp-admin/options-general.php(10):
require_once('...')
#9 {main}
  thrown
```

**Comment by ianhampton** [comment]
Thank you. The admin page was calling join() with the array to join followed by a string of the separator, this has been deprecated in PHP 8 and it is now expected that the separator will come first followed by the array to join/implode. I've just released version 2.1.16 to address this.

**Comment by mikeg-de** [comment]
Thanks a lot for your quick fix!
